pkg/util/ip/ip.go

```diff
@@ -226,3 +226,15 @@ func AppendPortIfMissing(addr, port string) string {
 
 	return net.JoinHostPort(addr, port)
 }
+
+func IsSubCIDR(parentNet, subNet *net.IPNet) bool {
+	parentMaskOnes, parentMaskBits := parentNet.Mask.Size()
+	subMaskOnes, subMaskBits := subNet.Mask.Size()
+	if parentMaskBits != subMaskBits {
+		return false
+	}
+	if parentMaskOnes >= subMaskOnes {
+		return false
+	}
+	return parentNet.Contains(subNet.IP) && subMaskOnes > parentMaskOnes
```

Comment:
In line 239, `subMaskOnes > parentMaskOnes` should be always true because of line 236 ?

Reply:
Does it work if subNet.IP is not the first IP in the cidr?

Reply:
> In line 239, `subMaskOnes > parentMaskOnes` should be always true because of line 236 ?

Thanks for reminding, you are right.

Reply:
> Does it work if subNet.IP is not the first IP in the cidr?

Like 192.168.1.255/24, 192.168.1.129/25? This can work.

Reply:
Added a ut for this.